src/CoreWCF.NetTcp/src/CoreWCF/Channels/Framing/NetMessageFramingConnectionHandler.cs

```diff
@@ -97,7 +98,22 @@ internal static UriPrefixTable<HandshakeDelegate> BuildAddressTable(IServiceProv
                         continue;
                     }
 
-                    HandshakeDelegate handshake = BuildHandshakeDelegateForDispatcher(dispatcher);
+                    IServiceDispatcher _serviceDispatcher = null;
+                    if (!(dispatcher.Binding is CustomBinding binding))
+                    {
+                        var _customBinding = new CustomBinding(dispatcher.Binding);
+                        if (_customBinding.Elements.Find<TcpTransportBindingElement>() != null)
+                        {
```

Comment:
Should be checking for ConnectionOrientedTransportBindingElement as this code is transport agnostic. It should eventually work as-is for Named Pipes and Unix Domain Sockets.

Reply:
added connectionoriented